e2e/fixtures/index.js

```diff
@@ -0,0 +1,82 @@
+const fs = require('fs')
+const Koa = require('koa')
+const path = require('path')
+
+const CURRENT_STATE_KEY = '__CURRENT__'
+const DEFAULT_STATE_KEY = '__DEFAULT__'
+
+const FIXTURE_SERVER_HOST = 'localhost'
+const FIXTURE_SERVER_PORT = 12345
+
+class FixtureServer {
+  constructor () {
+    this._app = new Koa()
+    this._stateMap = new Map([
+      [DEFAULT_STATE_KEY, Object.create(null)],
+    ])
+
+    this._app.use(async (ctx) => {
+      // Firefox is _super_ strict about needing CORS headers
+      ctx.set('Access-Control-Allow-Origin', '*')
+      if (this._isStateRequest(ctx)) {
+        ctx.body = this._stateMap.get(CURRENT_STATE_KEY)
+      }
+    })
+  }
+
+  async start () {
+    return new Promise((resolve) => {
```

Comment:
This should probably check for an existing `this._server` first, and stop that first. Or crash. Something like that.